### ISSUE TITLE
Video Scheduler Internal Error, BAR Above > 256 MiB

### ISSUE THREAD

**Issue by karfelren** [terminatorul/NvStrapsReBar#82]
**System**
 - Motherboard:Asus TUF FA506IH
 - BIOS Version:316
 - GPU: Asus GTX 1650 Mobile
 - [x] CSM is turned off. **Make sure to confirm this in the BIOS and not with GPU-Z or similar since it can be inaccurate**
 - [x] 4G decoding is enabled. **Make sure to confirm this in the BIOS and not with GPU-Z or similar since it can be inaccurate**
 - [ ] UEFIPatch is applied (see [Using UEFIPatch](https://github.com/xCuri0/ReBarUEFI/wiki/Using-UEFIPatch) for more information). On some motherboards [DSDT Patching](https://github.com/xCuri0/ReBarUEFI/wiki/DSDT-Patching) is also needed
 - [x] I have read [Common issues (and fixes)](https://github.com/xCuri0/ReBarUEFI/wiki/Common-issues-(and-fixes))

**Description**
This use version NvStrapsReBar v.03, To enable Above 4G Decoding with MMIO Limit 40bit I used UMAF. After a few minutes booting up windows will blue screen or sometimes randomly restart. 

I've tried adding Nvidia GOP "AMD and Nvidia GOP update (No requests, DIY)" but that just bricks the GPU.

![Screenshot (481)](https://github.com/terminatorul/NvStrapsReBar/assets/92556260/43d72ba6-e7b5-47f1-bcd8-2c803ff4113a)
![Screenshot (482)](https://github.com/terminatorul/NvStrapsReBar/assets/92556260/6ff1ad25-2ea7-464a-a231-1a9a2b34e78a)

**Comment by RipperPix00** [comment]
I also encountered the same problem, I disabled the iGPU and no longer have a blue screen. The iGPU is mentioned in the [Common issues (and fixes)](https://github.com/xCuri0/ReBarUEFI/wiki/Common-issues-(and-fixes)), but I didn't know how to fix it, so i had to disable it.

**Comment by RipperPix00** [comment]
There is another strange thing. Enabling rebar after i turn off the iGPU will cause the output of the Nvidia graphics card to change. When I don't enable rebar, the Nvidia graphics card can display the bios logo when the laptop power on, but after I enable rebar, there is no screen during the boot process, after a while the login screen of the system is directly displayed. This also menas that if I disable the iGPU, the bios menu cannot be displayed. 
My laptop's managent software originally had a menu to switch the graphics card output mode, after I disabled the iGPU an enabled rebar, this menu disappeared. 
Maybe rebar modified something after turing it on, which caused a conflict with the default Hybrid mode of the multiple GPUs, and then caused a blue screen, because the motherboard of the laptop has a chip to switch between the integrated graphics card and the discrete graphics card. I don't know if it can be solved in the future.

**Comment by karfelren** [comment]
> I also encountered the same problem, I disabled the iGPU and no longer have a blue screen. The iGPU is mentioned in the [Common issues (and fixes)](https://github.com/xCuri0/ReBarUEFI/wiki/Common-issues-(and-fixes)), but I didn't know how to fix it, so i had to disable it.

thanks, I will try it later, I don't have a monitor right now for testing

> There is another strange thing. Enabling rebar after i turn off the iGPU will cause the output of the Nvidia graphics card to change. When I don't enable rebar, the Nvidia graphics card can display the bios logo when the laptop power on, but after I enable rebar, there is no screen during the boot process, after a while the login screen of the system is directly displayed. This also menas that if I disable the iGPU, the bios menu cannot be displayed. My laptop's managent software originally had a menu to switch the graphics card output mode, after I disabled the iGPU an enabled rebar, this menu disappeared. Maybe rebar modified something after turing it on, which caused a conflict with the default Hybrid mode of the multiple GPUs, and then caused a blue screen, because the motherboard of the laptop has a chip to switch between the integrated graphics card and the discrete graphics card. I don't know if it can be solved in the future.

have you tried to set disable "Special Display Feature" and "Primary Video Adaptor" to Discrete GPU in BIOS?

**Comment by karfelren** [comment]
> Aw, man. I have a very similar laptop (i5-10300H, GTX 1650 Mobile), and NvStrapsReBar was working perfectly.
> 
> Today I started messing around in both Windows and the UEFI settings because the CPU wouldn't enter into C-States while on battery (not related to ReBar). I can't pinpoint the exact stuff I did, but perhaps some change related to PCIe APSD ended up ruining ReBar for this laptop and now I get these random “Video Scheduler Internal Error” BSODs. My laptop's screen is plugged to the iGPU, so I cannot disable it.
> 
> EDIT: Okay, hear me out. I solved it. While I was trying to find what could make the CPU not go into C-States, I remembered I had applied the “Disable NVIDIA HD audio device sleep timer” tweak from [NVCleanstall](https://www.techpowerup.com/download/techpowerup-nvcleanstall/). So, I thought I could revert the tweak and see if the CPU went idle afterwards. That's when the “Video Scheduler Internal Error” BSODs actually started. In my attempt to fix the BSODs, I installed the stock NVIDIA drivers, and that did not solve the issue. I reapplied the HD audio device sleep timer tweak through NVCleanstall and the BSODs disappeared.
> 
> Now, I'm wondering… how and why this sleep timer impacts resizable BAR?
> 
> **TLDR: Re-install the NVIDIA graphics using NVCleanstall, applying the “Disable NVIDIA HD audio device sleep timer” tweak.**

thanks! it resolve the problem and work perfectly (⁠≧⁠▽⁠≦⁠)
